src/main/java/frc/robot/subsystems/arm/ArmIOReal.java

```diff
@@ -123,235 +144,205 @@ public void setShoulderJointPower(double power) {
-        return !(elbowJoint.getX() < 0) || !(endPosition.getX() < 0);
-    }
-
-    public boolean armIsInRobot() {
-        return getEndPosition().getY() < 0 && getShoulderJointAngle().getDegrees() > 90;
-    }
-
-    public void setVelocity(Translation2d velocity) {
-        var armVelocities = kinematics.getVelocities(getEndPosition(), velocity);
-        setShoulderJointPower(armVelocities.shoulderAngle / Math.PI);
-        setElbowJointPower(armVelocities.elbowAngle / (2 * Math.PI));
-    }
-
-    public boolean changedToDefaultCommand() {
-        return changedToDefaultCommand;
-    }
-
-    /**
-     * Stops the motors.
-     */
-    public void stop() {
-        shoulderMainMotor.neutralOutput();
-        elbowMainMotor.neutralOutput();
-    }
-
-    @Override
-    public String getSubsystemName() {
-        return "Arm";
-    }
-
-    @Override
-    public void periodic() {
-        double t = Timer.getFPGATimestamp();
-        var currentCommand = getCurrentCommand();
-
-        changedToDefaultCommand = !(lastCommand instanceof ArmXboxControl || lastCommand instanceof ArmAxisXboxControlDumb)
-                && (currentCommand instanceof ArmXboxControl || currentCommand instanceof ArmAxisXboxControlDumb);
-        lastCommand = currentCommand;
-
-        if (currentCommand != null) {
-            Logger.getInstance().recordOutput("ArmCommand", currentCommand.getName());
+            /**
+             * Sets the angle of the elbow joint.
+             *
+             * @param angle desired angle. [degrees]
+             */
+            public void setElbowJointAngle ( double angle){
+                setElbowJointAngle(angle, 0, 0);
+            }
+
+            /**
+             * Sets the angle of the elbow joint.
+             *
+             * @param angle desired angle. [degrees]
+             */
+            public void setElbowJointAngle ( double angle, double ffMultiplier, int velocity){
+                angle = AngleUtil.normalize(angle);
+                Rotation2d error = new Rotation2d(angle).minus(new Rotation2d(currentShoulderAngle));
+                if (elbowEncoder.isConnected()) {
+                    motorControlRequest = new PositionVoltage(angle);
+                    elbowMainMotor.setControl(new PositionVoltage(angle, velocity, true, shoulderFeedforward * ffMultiplier, 1, true));
+
+                } else {
+                    elbowMainMotor.stopMotor();
+                }
+            }
+
+            /**
+             * Calculates the position of the end of the arm.
+             *
+             * @return Translation2d of the position.
+             */
+            public Translation2d getEndPosition () {
+                double shoulderAngle = getShoulderJointAngle().getRadians();
+                double elbowAngle = getElbowJointAngle().getRadians();
+                return kinematics.forwardKinematics(shoulderAngle, shoulderAngle + elbowAngle - Math.PI);
```

Comment:
`kinematics` should accept a `Rotation2d` or use [the new unit system](https://docs.wpilib.org/en/latest/docs/software/basic-programming/java-units.html).